grpc_opa/authorizer.go

```diff
@@ -172,50 +172,50 @@ func (a *DefaultAuthorizer) Evaluate(ctx context.Context, fullMethod string, req
 		return false, ctx, fmt.Errorf("%q", errs)
 	}
 
-	req := Payload{
+	opaReq := Payload{
 		Endpoint:    parseEndpoint(fullMethod),
 		FullMethod:  fullMethod,
 		Application: a.application,
 		// FIXME: implement athena_claims.AuthBearersFromCtx
 		JWT: redactJWT(rawJWT),
 	}
 
-	decisionInput, err := a.decisionInputHandler.GetDecisionInput(ctx, fullMethod, request)
+	decisionInput, err := a.decisionInputHandler.GetDecisionInput(ctx, fullMethod, grpcReq)
 	if decisionInput == nil || err != nil {
 		logger.WithFields(log.Fields{
 			"fullMethod": fullMethod,
 		}).WithError(err).Error("get_decision_input")
 		return false, ctx, ErrInvalidArg
 	}
 	//logger.Debugf("decisionInput=%+v", *decisionInput)
-	req.DecisionInput = *decisionInput
+	opaReq.DecisionInput = *decisionInput
 
-	reqJSON, err := json.Marshal(req)
+	opaReqJSON, err := json.Marshal(opaReq)
 	if err != nil {
 		logger.WithFields(log.Fields{
-			"request": request,
-		}).WithError(err).Error("request_json_marshal")
+			"opaReq": opaReq,
+		}).WithError(err).Error("opa_request_json_marshal")
 		return false, ctx, ErrInvalidArg
 	}
 
 	now := time.Now()
 	logger.WithFields(log.Fields{
-		"request": request,
-		//"requestJSON": string(reqJSON),
-	}).Debug("authorization_request")
+		"opaReq": opaReq,
+		//"opaReqJSON": string(opaReqJSON),
+	}).Debug("opa_authorization_request")
 
 	// To enable tracing, the context must have a tracer attached
 	// to it. See the tracing documentation on how to do this.
 	ctx, span := trace.StartSpan(ctx, fmt.Sprint(SERVICENAME, fullMethod))
 	{
 		span.Annotate([]trace.Attribute{
-			trace.StringAttribute("in", string(reqJSON)),
+			trace.StringAttribute("in", string(opaReqJSON)),
 		}, "in")
 	}
 	// FIXME: perhaps only inject these fields if this is the default handler
 
-	var response OPAResponse
-	err = opaEvaluator(ctxlogrus.ToContext(ctx, logger), decisionInput.DecisionDocument, request, &response)
+	var opaResp OPAResponse
+	err = opaEvaluator(ctxlogrus.ToContext(ctx, logger), decisionInput.DecisionDocument, opaReq, &opaResp)
```

Comment:
This is the actual ATLAS-10442 bug fix.

Reply:
There are a lot of changes for a one-line fix.